monitoring/nethealth_test.go

```diff
@@ -0,0 +1,211 @@
+			},
+			incomingData: []networkData{
+				networkData{requestTotal: 1, timeoutTotal: 1},
+				networkData{requestTotal: 2, timeoutTotal: 2},
+				networkData{requestTotal: 3, timeoutTotal: 3},
+				networkData{requestTotal: 4, timeoutTotal: 4},
+				networkData{requestTotal: 5, timeoutTotal: 5},
+			},
+		},
+		{
+			comment: Commentf("Expected oldest data point to be removed"),
+			expected: peerData{
+				prevRequestTotal: 7,
+				prevTimeoutTotal: 2,
+				requestInc:       []float64{1, 1, 1, 1, 2},
+				timeoutInc:       []float64{0, 0, 0, 0, 1},
+			},
+			incomingData: []networkData{
+				networkData{requestTotal: 1, timeoutTotal: 1},
+				networkData{requestTotal: 2, timeoutTotal: 1},
+				networkData{requestTotal: 3, timeoutTotal: 1},
+				networkData{requestTotal: 4, timeoutTotal: 1},
+				networkData{requestTotal: 5, timeoutTotal: 1},
+				networkData{requestTotal: 7, timeoutTotal: 2},
+			},
+		},
+	}
+
+	for _, testCase := range testCases {
+		testCase := testCase
+		checker := s.newNethealthChecker()
+
+		for _, data := range testCase.incomingData {
+			c.Assert(checker.updatePeer(testNode, data), IsNil, testCase.comment)
+		}
+
+		data, err := checker.peerStats.Get(testNode)
+		c.Assert(err, IsNil, testCase.comment)
+		c.Assert(data, test.DeepCompare, testCase.expected, testCase.comment)
+	}
+}
+
+// TestNethealthChecker verifies nethealth checker can properly detect
+// healthy/unhealthy network.
+func (s *NethealthSuite) TestNethealthVerification(c *C) {
+	var testCases = []struct {
+		comment    CommentInterface
+		expected   health.Reporter
+		packetLoss []float64
+		storedData peerData
+	}{
+		{
+			comment:  Commentf("Expected no failed probes. Not enough data points."),
+			expected: new(health.Probes),
+			storedData: peerData{
+				requestInc: []float64{10, 20, 30},
+				timeoutInc: []float64{5, 10, 15},
+			},
+		},
+		{
+			comment:  Commentf("Expected no failed probes. No timeouts."),
+			expected: new(health.Probes),
+			storedData: peerData{
+				requestInc: []float64{10, 20, 30, 40, 50},
+				timeoutInc: []float64{0, 0, 0, 0, 0},
+			},
+		},
+		{
+			comment:  Commentf("Expected no failed probes. Data points do not exceed threshold."),
+			expected: new(health.Probes),
+			storedData: peerData{
+				requestInc: []float64{10, 20, 30, 40, 50},
+				timeoutInc: []float64{2, 4, 6, 8, 10},
+			},
+		},
+		{
+			comment:  Commentf("Expected no failed probes. Does not exceed threshold throughout entire interval."),
+			expected: new(health.Probes),
+			storedData: peerData{
+				requestInc: []float64{10, 20, 30, 40, 50},
+				timeoutInc: []float64{10, 20, 30, 40, 0},
+			},
+		},
+		{
+			comment:  Commentf("Expected failed probe. Timeouts increase at each interval."),
+			expected: &health.Probes{NewProbeFromErr(nethealthCheckerID, nethealthDetail(testNode), nil)},
+			storedData: peerData{
+				requestInc: []float64{10, 20, 30, 40, 50},
+				timeoutInc: []float64{10, 20, 30, 40, 50},
+			},
```

Comment:
You don't exercise the length difference test between requests/timeouts which is handled in code.

Reply:
1c9ec8b
Added some test cases for parseMetrics.